spec/controllers/miq_ae_customization_controller/dialogs_spec.rb

```diff
@@ -79,10 +79,44 @@
       it "Dialog with out Dialog fields should not be saved" do
         allow(controller).to receive(:render_flash)
         controller.send(:dialog_edit)
-        expect(assigns(:flash_array).first[:message]).to include("Dialog must have at least one Element")
+        expect(assigns(:flash_array).first[:message]).to include("Dialog a1 must have at least one Tab")
       end
 
-      it "Adds a Dialog with Tab/Groups/Field" do
+      it "Any Dialog/Tab with empty group should not be added, and display multiple message for each empty group" do
+        new_hash = {
+          :label       => "Dialog 1",
+          :description => "Dialog 1",
+          :buttons     => ["submit"],
+          :tabs        => [
+            {
+              :label       => "Tab 1",
+              :description => "Tab 1",
+              :groups      => [
+                {
+                  :label       => "Box 1",
+                  :description => "Box 1",
+                  :fields      => []
+                },
+                {
+                  :label       => "Box 2",
+                  :description => "Box 2",
+                  :fields      => []
+                }
+              ]
+            }
+          ]
+        }
+        controller.instance_variable_set(:@lastaction, "replace_right_cell")
+        assigns(:edit)[:new] = new_hash
+        controller.stub(:get_node_info)
+        controller.stub(:replace_right_cell)
+        controller.stub(:render_flash)
+        controller.instance_variable_set(:@_params, :button => "add")
+        controller.send(:dialog_edit)
+        assigns(:flash_array).first[:message].should include("Validation failed: Dialog Dialog 1 / Tab Tab 1 / Box Box 1 must have at least one Element, Dialog Dialog 1 / Tab Tab 1 / Box Box 2 must have at least one Element")
```

Comment:
Please change to use the `expect` syntax.  Would also be good to break this into multiple lines.